axelrod/tournament.py

```diff
@@ -48,19 +48,23 @@ def __init__(self, players, match_generator=RoundRobinMatches,
             self.game = game
         self.players = players
         self.repetitions = repetitions
-        if deterministic_cache is None:
-            self.deterministic_cache = DeterministicCache()
-        else:
-            self.deterministic_cache = deterministic_cache
-        self.match_generator = match_generator(
-            players, turns, self.game, self.deterministic_cache)
+        self.match_generator = match_generator(players, turns, self.game,
+                                               self.repetitions)
         self._with_morality = with_morality
         self._parallel_repetitions = repetitions
         self._processes = processes
         self._logger = logging.getLogger(__name__)
-        self.interactions = []
+        self.interactions = defaultdict(list)
 
-    def play(self, filename=None):
+        if not filename:
+            self.outputfile = NamedTemporaryFile(mode='w')
+            filename = self.outputfile.name
+        else:
+            self.outputfile = open(filename, 'w')
+        self.writer = csv.writer(self.outputfile)
+        self.filename = filename
+
+    def play(self):
```

Comment:
Perhaps:


```
def play(self, build_results=True):
    ...
```


So that you can pass `build_results=False` then you don't build the results set (an error/warning would be raised if no filename is given and `build_results` is set to `True`)? Just for the case that you'd want the data but not actually want to analyse it (for example my final year student this year was analysing the results with R so there would not have been the need for the higher memory consumption for the Results set).